### ISSUE TITLE
invalid byte sequence in UTF-8 (ArgumentError)

### ISSUE THREAD

**Issue by shivabhusal** [tilo/smarter_csv#103]
```
/home/john/.rvm/gems/ruby-2.4.0/gems/smarter_csv-1.1.0/lib/smarter_csv/smarter_csv.rb:32:in `sub': invalid byte sequence in UTF-8 (ArgumentError)
	from /home/john/.rvm/gems/ruby-2.4.0/gems/smarter_csv-1.1.0/lib/smarter_csv/smarter_csv.rb:32:in `process'
	from /home/john/projects/hulaki/lib/hulaki/contact_parser.rb:7:in `perform'
	from /home/john/projects/hulaki/lib/hulaki/search_engine.rb:8:in `initialize'
	from /home/john/projects/hulaki/lib/hulaki/core.rb:76:in `new'
	from /home/john/projects/hulaki/lib/hulaki/core.rb:76:in `handle_search_and_clipboard'
	from /home/john/projects/hulaki/lib/hulaki/core.rb:10:in `perform'
	from /home/john/projects/hulaki/exe/hulaki:10:in `<top (required)>'
	from /home/john/.rvm/gems/ruby-2.4.0/bin/hulaki:23:in `load'
	from /home/john/.rvm/gems/ruby-2.4.0/bin/hulaki:23:in `<main>'
	from /home/john/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `eval'
	from /home/john/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `<main>'

```

Setting breakpoints

```ruby
From: /home/john/.rvm/gems/ruby-2.4.0/gems/smarter_csv-1.1.0/lib/smarter_csv/smarter_csv.rb @ line 33 SmarterCSV.process:

    28: 
    29:       if options[:headers_in_file]        # extract the header line
    30:         # process the header line in the CSV file..
    31:         # the first line of a CSV file contains the header .. it might be commented out, so we need to read it anyhow
    32:         
 => 33:         require 'pry';binding.pry
    34:       
    35:         header = f.readline.sub(options[:comment_regexp],'').chomp(options[:row_sep])
    36:         line_count += 1
    37:         header = header.gsub(options[:strip_chars_from_headers], '') if options[:strip_chars_from_headers]
    38:         if (header =~ %r{#{options[:quote_char]}}) and (! options[:force_simple_split])

[1] pry(SmarterCSV)> f.readline
=> "\xFF\xFEN\u0000a\u0000m\u0000e\u0000,\u0000G\u0000i\u0000v\u0000e\u0000n\u0000 \u0000N\u0000a\u0000m\u0000e\u0000,\u0000A\u0000d\u0000d\u0000i\u0000t\u0000i\u0000o\u0000n\u0000a\u0000l\u0000 \u0000N\u0000a\u0000m\u0000e\u0000,\u0000F\u0000a\u0000m\u0000i\u0000l\u0000y\u0000 \u0000N\u0000a\u0000m\u0000e\u0000,\u0000Y\u0000o\u0000m\u0000i\u0000 \u0000N\u0000a\u0000m\u0000e\u0000,\u0000G\u0000i\u0000v\u0000e\u0000n\u0000 \u0000N\u0000a\u0000m\u0000e\u0000 \u0000Y\u0000o\u0000m\u0000i\u0000,\u0000A\u0000d\u0000d\u0000i\u0000t\u0000i\u0000o\u0000n\u0000a\u0000l\u0000 \u0000N\u0000a\u0000m\u0000e\u0000 \u0000Y\u0000o\u0000m\u0000i\u0000,\u0000F\u0000a\u0000m\u0000i\u0000l\u0000y\u0000 \u0000N\u0000a\u0000m\u0000e\u0000 \u0000Y\u0000o\u0000m\u0000i\u0000,\u0000N\u0000a\u0000m\u0000e\u0000 \u0000P\u0000r\u0000e\u0000f\u0000i\u0000x\u0000,\u0000N\u0000a\u0000m\u0000e\u0000 \u0000S\u0000u\u0000f\u0000f\u0000i\u0000x\u0000,\u0000I\u0000n\u0000i\u0000t\u0000i\u0000a\u0000l\u0000s\u0000,\u0000N\u0000i\u0000c\u0000k\u0000n\u0000a\u0000m\u0000e\u0000,\u0000S\u0000h\u0000o\u0000r\u0000t\u0000 \u0000N\u0000a\u0000m\u0000e\u0000,\u0000M\u0000a\u0000i\u0000d\u0000e\u0000n\u0000 \u0000N\u0000a\u0000m\u0000e\u0000,\u0000B\u0000i\u0000r\u0000t\u0000h\u0000d\u0000a\u0000y\u0000,\u0000G\u0000e\u0000n\u0000d\u0000e\u0000r\u0000,\u0000L\u0000o\u0000c\u0000a\u0000t\u0000i\u0000o\u0000n\u0000,\u0000B\u0000i\u0000l\u0000l\u0000i\u0000n\u0000g\u0000 \u0000I\u0000n\u0000f\u0000o\u0000r\u0000m\u0000a\u0000t\u0000i\u0000o\u0000n\u0000,\u0000D\u0000i\u0000r\u0000e\u0000c\u0000t\u0000o\u0000r\u0000y\u0000 \u0000S\u0000e\u0000r\u0000v\u0000e\u0000r\u0000,\u0000M\u0000i\u0000l\u0000e\u0000a\u0000g\u0000e\u0000,\u0000O\u0000c\u0000c\u0000u\u0000p\u0000a\u0000t\u0000i\u0000o\u0000n\u0000,\u0000H\u0000o\u0000b\u0000b\u0000y\u0000,\u0000S\u0000e\u0000n\u0000s\u0000i\u0000t\u0000i\u0000v\u0000i\u0000t\u0000y\u0000,\u0000P\u0000r\u0000i\u0000o\u0000r\u0000i\u0000t\u0000y\u0000,\u0000S\u0000u\u0000b\u0000j\u0000e\u0000c\u0000t\u0000,\u0000N\u0000o\u0000t\u0000e\u0000s\u0000,\u0000G\u0000r\u0000o\u0000u\u0000p\u0000 \u0000M\u0000e\u0000m\u0000b\u0000e\u0000r\u0000s\u0000h\u0000i\u0000p\u0000,\u0000E\u0000-\u0000m\u0000a\u0000i\u0000l\u0000 \u00001\u0000 \u0000-\u0000 \u0000T\u0000y\u0000p\u0000e\u0000,\u0000E\u0000-\u0000m\u0000a\u0000i\u0000l\u0000 \u00001\u0000 \u0000-\u0000 \u0000V\u0000a\u0000l\u0000u\u0000e\u0000,\u0000E\u0000-\u0000m\u0000a\u0000i\u0000l\u0000 \u00002\u0000 \u0000-\u0000 \u0000T\u0000y\u0000p\u0000e\u0000,\u0000E\u0000-\u0000m\u0000a\u0000i\u0000l\u0000 \u00002\u0000 \u0000-\u0000 \u0000V\u0000a\u0000l\u0000u\u0000e\u0000,\u0000P\u0000h\u0000o\u0000n\u0000e\u0000 \u00001\u0000 \u0000-\u0000 \u0000T\u0000y\u0000p\u0000e\u0000,\u0000P\u0000h\u0000o\u0000n\u0000e\u0000 \u00001\u0000 \u0000-\u0000 \u0000V\u0000a\u0000l\u0000u\u0000e\u0000,\u0000P\u0000h\u0000o\u0000n\u0000e\u0000 \u00002\u0000 \u0000-\u0000 \u0000T\u0000y\u0000p\u0000e\u0000,\u0000P\u0000h\u0000o\u0000n\u0000e\u0000 \u00002\u0000 \u0000-\u0000 \u0000V\u0000a\u0000l\u0000u\u0000e\u0000,\u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000 \u00001\u0000 \u0000-\u0000 \u0000T\u0000y\u0000p\u0000e\u0000,\u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000 \u00001\u0000 \u0000-\u0000 \u0000F\u0000o\u0000r\u0000m\u0000a\u0000t\u0000t\u0000e\u0000d\u0000,\u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000 \u00001\u0000 \u0000-\u0000 \u0000S\u0000t\u0000r\u0000e\u0000e\u0000t\u0000,\u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000 \u00001\u0000 \u0000-\u0000 \u0000C\u0000i\u0000t\u0000y\u0000,\u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000 \u00001\u0000 \u0000-\u0000 \u0000P\u0000O\u0000 \u0000B\u0000o\u0000x\u0000,\u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000 \u00001\u0000 \u0000-\u0000 \u0000R\u0000e\u0000g\u0000i\u0000o\u0000n\u0000,\u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000 \u00001\u0000 \u0000-\u0000 \u0000P\u0000o\u0000s\u0000t\u0000a\u0000l\u0000 \u0000C\u0000o\u0000d\u0000e\u0000,\u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000 \u00001\u0000 \u0000-\u0000 \u0000C\u0000o\u0000u\u0000n\u0000t\u0000r\u0000y\u0000,\u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000 \u00001\u0000 \u0000-\u0000 \u0000E\u0000x\u0000t\u0000e\u0000n\u0000d\u0000e\u0000d\u0000 \u0000A\u0000d\u0000d\u0000r\u0000e\u0000s\u0000s\u0000,\u0000W\u0000e\u0000b\u0000s\u0000i\u0000t\u0000e\u0000 \u00001\u0000 \u0000-\u0000 \u0000T\u0000y\u0000p\u0000e\u0000,\u0000W\u0000e\u0000b\u0000s\u0000i\u0000t\u0000e\u0000 \u00001\u0000 \u0000-\u0000 \u0000V\u0000a\u0000l\u0000u\u0000e\u0000\r\u0000\n"
[2] pry(SmarterCSV)> options[:comment_regexp]
=> /^#/


```

**Comment by contentfree** [comment]
Similar bug happens in 1.1.4. At line 42, the `f.readline` blows up because of the invalid byte sequence. It doesn't reach the next line where it tries to force/fix the encoding. What do?

**Comment by contentfree** [comment]
The same pattern exists at line 102. The `f.readline` blows up before even getting a change to force encoding.

**Comment by coryrmathis** [comment]
Has anyone found a workaround to this.  My client often tries to import CSV files that are in a Windows encoding and this bug makes things very complicated.

**Comment by tilo** [comment]
fixed in 1.1.5
